src/main/frontend/components/reference.cljs

```diff
@@ -72,7 +72,7 @@
   [page-name n-ref]
   (let [ref-blocks (db/get-page-unlinked-references page-name)
         encoded-page-name (util/url-encode page-name)]
-    (reset! n-ref (count ref-blocks))
+    (js/setTimeout #(reset! n-ref (count ref-blocks)) 200)
```

Comment:
We can add a `nil?` check to avoid the warning, a better way is to use `rum/use-effect?` that sadly our current implementation doesn't support.
```
(when (nil? @n-ref)
      (reset! n-ref (count ref-blocks)))
```

Reply:
The warning is related to the other lazy-foldable problem. If we pass the fn, the warning will disappear. The warning is: 

```
Warning: Cannot update during an existing state transition (such as within `render`). Render methods should be a pure function of props and state.
```

add a `nil?` checker seems not working also.

Reply:
We can try this:
``` clj
(rum/defcs unlinked-references-aux < rum/reactive db-mixins/query
  {:will-mount (fn [state]
                 (let [[page-name n-ref] (:rum/args state)
                       ref-blocks (db/get-page-unlinked-references page-name)]
                   (reset! n-ref (count ref-blocks))
                   (assoc state ::ref-blocks ref-blocks)))}
  [state page-name n-ref]
  (let [ref-blocks (::ref-blocks state)]
    [:div.references-blocks
     (let [ref-hiccup (block/->hiccup ref-blocks
                                      {:id (str page-name "-unlinked-")
                                       :start-level 2
                                       :ref? true
                                       :group-by-page? true
                                       :editor-box editor/box}
                                      {})]
       (content/content page-name
                        {:hiccup ref-hiccup}))]))
```

Reply:
It moves the side effects to the `will-mount` lifecycle instead of the rendering body.

Reply:
Good!

By the way,  the will-mount API  seems to be deprecated by react official. Will we replace it with another solution in the future?

Reply:
Yep, I think so.